src/esm_runscripts/config_control.py

```diff
@@ -0,0 +1,112 @@
+import yaml
+import esm_parser
+
+
+def check_configuration_consistency(config):
+    """ TODO: fill docstring
+    """
+    runscript_abspath = config["general"]["command_line_config"]["runscript_abspath"]
+    # TODO: can this be None or KeyError?
+    models = config['general']['valid_model_names']
+
+    with open(runscript_abspath, "rt") as f:
+        runscript = yaml.load(f, Loader=yaml.FullLoader)
+
+    for model in models:
+        model_config_path = config[model]["debug_info"]["loaded_from_file"]
+        # some models have multiple files. Take the one with <model>.yaml
+        if isinstance(model_config_path, list):
+            model_config_path = model_config_path[0]
+            # deniz: remaining lines was a good idea but don't always work
+            # base_name = f"{model}.yaml"
+            # model_config_path = next((item for item in model_config_path if base_name in item), None)
+        with open(model_config_path, "rt") as f:
+            model_config = yaml.load(f, Loader=yaml.FullLoader)
+
+        check_version(config, model_config, runscript, model)
```

Comment:
What happens with the return values? In your functions, you give back true but that is never used.